modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/UserCardTag.java

```diff
@@ -0,0 +1,512 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.frontend.taglib.clay.servlet.taglib.soy.UserCard;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItem;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.LabelItem;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Carlos Lancha
+ */
+public class UserCardTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		if (_userCard != null) {
+			_populate();
```

Comment:
Do we really need the `populate`?

One option we could avoid it is by just using our `getters` internally and implementing this like:

```
public String getIcon() {
    if (_icon == null && _userCard != null) {
        return _userCard.getIcon();
    }

    return _icon;
}

Reply:
I followed the `populate` method approach because it's how it's being done in other tags. I don't really have a strong opinion here, just that I'd prefer to use the same pattern everywhere.

Reply:
@jbalsas Wouldn't that approach make order of attributes important?

`_userCard` is null on check:
```
<clay:user-card-v2
	icon="foo"
	userCard="<%= new UserCard(...) %>"
/>
```
vs
`_userCard` is not null on check:
```
<clay:user-card-v2
        userCard="<%= new UserCard(...) %>"
	icon="foo"
/>
```

By setting `populate` in `doStartTag` we avoid this.

Reply:
Oh it's a getter, nevermind me

Reply:
> I followed the populate method approach because it's how it's being done in other tags. I don't really have a strong opinion here, just that I'd prefer to use the same pattern everywhere.

It was used in the `soy` tags, because they do populate a context map. That's not the case for our new tags, so we don't need to pre-process it like that.

Reply:
Oki, I'll change it

Reply:
@jbalsas @carloslancha I ran into a bit of a problem with not setting `cssClass` in `populate` and using extended getter. As we never set the value of `_cssClass`, it is null in the `BaseContainerTag` `processCssClasses`:
```
protected String processCssClasses(Set<String> cssClasses) {
	if (Validator.isNotNull(_cssClass)) {
		cssClasses.addAll(StringUtil.split(_cssClass, CharPool.SPACE));
	}

	return StringUtil.merge(cssClasses, StringPool.SPACE);
}
```
I see these solutions:
1. Replace  `_cssClass` with `getCssClass()` in `BaseContainerTag` , and do the same for all other `_*` vars (there is a lot of them).
1. Revert to `_populate` and explicit setting of properties in `CardTag` class. 
1. Overwrite `processCssClasses` in `CardTag` class. This is probably the worst solution, as we would be duplicating parsing.

Reply:
I'd go with the first one, we just need to replace every `_whatever` in `BaseContainerTag` with `getWhatever()`, right?

Reply:
I believe we would only need to do this for properties settable in model pojo (`id` and `cssClass`), so it's not that bad. I'm not sure if that would break some backend convention, as we would inconsistently use` _someAttr` and `getSomeOtherAttr()` in the class. It's probably fine...